tests/checkBoxes.spec.ts

```diff
@@ -0,0 +1,137 @@
+    // Edit the fourth veterinarian in the list
+    await page.getByText('Edit Vet').nth(3).click()
+
+    /* 3. Add assertion that "Specialties" field is displayed value "surgery" */
+    const specialtiesDropdown = page.locator('div.dropdown')
+    const specialtiesDropdownText = page.locator('.selected-specialties')
+
+    // Validate that the specialties listed is only 'surgery'
+    await expect(specialtiesDropdownText).toHaveText('surgery')
+
+    /* 4. Click on the "Specialties" drop-down menu */
+    // Click on the specialties dropdown to show options
+    await specialtiesDropdown.click()
+
+    /* 5. Check all specialties from the list */
+    /* 6. Add assertion that all specialties are checked */
+
+    // Capture all specialties from checkbox
+    const specialties = page.locator('.dropdown-content label')
+    
+    for(const specialty of await specialties.allTextContents()){
+        const checkbox = page.getByRole('checkbox', {name: specialty})
+        
+        // Check every box
+        await checkbox.check()
+
+        // Validate that the box is checked
+        await expect(checkbox).toBeChecked()
+    }
+
+    /* 7. Add assertion that all checked specialities are displayed in the "Specialties" field */
+    // Validate that all specialties are displayed accordingly
+    await expect(specialtiesDropdownText).toHaveText('surgery, radiology, dentistry, nail trims')
+});
+
+test('Unselect all specialties', async ({page}) => {
+    /* 2. Select the third veterinarian "Linda Douglas" and click "Edit Vet" button */
+    // Edit the third veterinarian in the list
+    await page.locator('button', {hasText: 'Edit Vet'}).nth(2).click()
+
+    /* 3. Add assertion of the "Specialties" field displayed value "surgery, dentistry" */
+    const specialtiesDropdown = page.locator('div.dropdown')
+    const specialtiesDropdownText = page.locator('div span.selected-specialties')
+
+    // Validate that the dropdown displays both 'dentistry' and 'surgery' selections
+    await expect(specialtiesDropdownText).toHaveText('dentistry, surgery')
+    
+    /* 4. Click on the "Specialties" drop-down menu */
+    await specialtiesDropdown.click()
+
+    /* 5. Uncheck all specialties from the list */
+    /* 6. Add assertion that all specialties are unchecked */
+
+    // Capture all specialties from checkbox
+    const specialties = page.locator('.dropdown-content label')
+    
+    for(const specialty of await specialties.allTextContents()){
+        const checkbox = page.getByRole('checkbox', {name: specialty})
+
+        // Check every box
+        await checkbox.uncheck()
+
+        // Validate that the box is unchecked
+        await expect(checkbox).not.toBeChecked()
+    }
+
+    /* 7. Add assertion that "Specialties" field is empty */
+    // Validate that there are no specialties listed and the dropdown is empty
+    await expect(specialtiesDropdownText).toHaveText('')
```

Comment:
Locator Assertion has a better locator to validate if there is no value rather than validating the empty string value.
You can find this example in the blog post I have shared with you earlier.